ubuntu-autoinstall-generator.sh

```diff
@@ -213,11 +236,22 @@ chmod -R u+w "$tmpdir"
 rm -rf "$tmpdir/"'[BOOT]'
 log "👍 Extracted to $tmpdir"
 
+if [ ${focal} -eq 0 ]; then
+        log "🔧 Extracting EFI images from image..."
+        efi_start=$(fdisk -o Start,Type -l "${source_iso}" | grep -oP '\d+(?=\s+EFI-System)')
+        efi_length=$(fdisk -o Sectors,Type -l "${source_iso}" | grep -oP '\d+(?=\s+EFI-System)')
```

Comment:
This did not work for me. `fdisk` (version 2.37.2) output is:

```
Disk ubuntu-22.04-live-server-amd64.iso: 1.37 GiB, 1466714112 bytes, 2864676 sectors
Units: sectors of 1 * 512 = 512 bytes
Sector size (logical/physical): 512 bytes / 512 bytes
I/O size (minimum/optimal): 512 bytes / 512 bytes
Disklabel type: gpt
Disk identifier: C84E0225-4BE7-447A-9FA1-EBF040BDC01F

  Start Type
     64 Microsoft basic data
2855516 EFI System
2864012 Microsoft basic data
```

Hence:
```suggestion
        efi_start=$(fdisk -o Start,Type -l "${source_iso}" | grep -oP '\d+(?=\s+EFI System)')
        efi_length=$(fdisk -o Sectors,Type -l "${source_iso}" | grep -oP '\d+(?=\s+EFI System)')
```

Reply:
I suppose that might be a locale issue. I will verify later today. I guess "EFI.System" should work in most cases

Reply:
These 2 `fdisk` lines don't return anything for me. `fdisk from util-linux 2.37.2` On Ubuntu 22.04

`fdisk` of the ISO for me is :

```
➭ fdisk -o Sectors,Type -l ubuntu-22.04-live-server-amd64.iso                                  
Disk ubuntu-22.04-live-server-amd64.iso: 1.37 GiB, 1466714112 bytes, 2864676 sectors
Units: sectors of 1 * 512 = 512 bytes
Sector size (logical/physical): 512 bytes / 512 bytes
I/O size (minimum/optimal): 512 bytes / 512 bytes
Disklabel type: gpt
Disk identifier: C84E0225-4BE7-447A-9FA1-EBF040BDC01F

Sectors Type
2855452 Microsoft basic data
   8496 EFI System
    600 Microsoft basic data


➭ fdisk -o Start,Type -l ubuntu-22.04-live-server-amd64.iso
Disk ubuntu-22.04-live-server-amd64.iso: 1.37 GiB, 1466714112 bytes, 2864676 sectors
Units: sectors of 1 * 512 = 512 bytes
Sector size (logical/physical): 512 bytes / 512 bytes
I/O size (minimum/optimal): 512 bytes / 512 bytes
Disklabel type: gpt
Disk identifier: C84E0225-4BE7-447A-9FA1-EBF040BDC01F

  Start Type
     64 Microsoft basic data
2855516 EFI System
2864012 Microsoft basic data
```

I needed to use the `EFI.System` also to generate an ISO.

Generated ISO worked great when used as an IPMI virtual media (Supermicro server).

Used `dd` to copy the ISO to a thumb drive and... It worked !!! Great job @eNBeWe !

Reply:
I just checked on my system and I definitely have the dash in there. So I changed it to the more general RegExp with "EFI.System".

Happy to hear that it seems to work for y'all